includes/KSES.php

```diff
@@ -26,20 +26,46 @@
 
 namespace Google\Web_Stories;
 
+use WP_Post_Type;
+
 /**
  * KSES class.
  *
  * Provides KSES utility methods to override the ones from core.
  */
 class KSES {
 	/**
-	 * Initializes KSES filters for stories.
+	 * Initializes KSES filters for stories only if capability check are met.
 	 *
 	 * @since 1.0.0
 	 *
 	 * @return void
 	 */
 	public function init() {
+
+		$create_posts     = false;
+		$post_type_object = get_post_type_object( Story_Post_Type::POST_TYPE_SLUG );
+		if (
+			$post_type_object instanceof WP_Post_Type &&
+			property_exists( $post_type_object->cap, 'create_posts' )
+		) {
+
+			$create_posts = current_user_can( $post_type_object->cap->create_posts );
+		}
+
+		if ( $create_posts ) {
+			$this->add_filters();
+		}
```

Comment:
The confusing part about conditionally adding these filters is that we check whether the user an edit stories, but the filters actually apply to all post types.

Hmm 🤔

Reply:
Why give a user that does have access to edit stories these capabilities? The reason we add these filter so you can publish web stories, if you can't create web stories, there is no point.

Reply:
Also it worth noting, init was poorly named in the first place. init methods normally map to add action init. Add filters is much cleaner. Now that init used to hook into add action, this is make sense to me. It just changing the name of the method to something clearer. I can revert if it is a problem, but what do you call the init function?

Reply:
All I am saying is that this is confusing:

> If user can edit _stories_, allow user to use more markup in _all post types_

I am not saying it is _bad_ or anything. Just _confusing_, because there simply isn't another way (as we have found out).

> Also it worth noting, init was poorly named in the first place. init methods normally map to add action init.

FWIW this was named after `kses_init()`. I am not suggesting to rename the method now.